src/session.ts

```diff
@@ -665,6 +689,29 @@ export class Session extends EventEmitter {
 
     let url = this.url
 
+    if (this.customUrlHandler && !this.url) {
+      const customUrlHandlerPath = path.resolve(
+        process.cwd(),
+        this.customUrlHandler,
+      )
+      if (!fs.existsSync(customUrlHandlerPath)) {
+        console.error(`Custom error handler not found: ${customUrlHandlerPath}`)
+        process.exit(1)
+      }
+      const customUrlHandler = (
+        await import(/* webpackIgnore: true */ customUrlHandlerPath)
+      ).default
+
+      url = customUrlHandler({
```

Comment:
I would mark the `customUrlHandler` function as async, just to cover all the possible use cases.